cli/cmd/install.go

```diff
@@ -346,11 +350,11 @@ control plane.`,
 	return cmd
 }
 
-func installRunE(options *installOptions, stage string, flags *pflag.FlagSet) error {
-	if !options.ignoreCluster {
-		// TODO: consider cobra.SilenceUsage, so we can return errors from
-		// `RunE`, rather than calling `os.Exit(1)`
-		exitIfClusterExists()
+func installRunE(options *installOptions, stage string, flags *pflag.FlagSet, preInstallCheck func() bool) error {
+	controlPlaneExists := preInstallCheck()
```

Comment:
`controlPlaneExists` isn't quite accurate, as `preInstallCheck()` has different meanings depending on the caller.

consider moving the call to `preInstallCheck()` (and related `!options.ignoreCluster` logic) out of `installRunE`. given that we only have 3 cases (`install`, `install config`, `install control-plane`), and each is sufficiently different, i think it's helpful to put this logic explicitly into the respective `newCmdInstall*()` functions.